src/test/java/io/confluent/examples/streams/GenericAvroIntegrationTest.java

```diff
@@ -80,10 +77,11 @@ public void shouldRoundTripGenericAvroDataThroughKafka() throws Exception {
 
     final Properties streamsConfiguration = new Properties();
     streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, "generic-avro-integration-test");
-    streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
+    streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy config");
     streamsConfiguration.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.ByteArray().getClass().getName());
-    streamsConfiguration.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, GenericAvroSerde.class);
-    streamsConfiguration.put(AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, CLUSTER.schemaRegistryUrl());
+    streamsConfiguration.put("mock.schema.registry.client", mockSchemaRegistryClient);
+    streamsConfiguration.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, MockGenericAvroSerde.class);
```

Comment:
This is a sticky point. The GenericAvroSerde class (and company) are totally incompatible with providing a MockSchemaRegistryClient. Even leaving aside passing one vial the `configure` method, they have only package-private constructors that accept a pre-constructed client.

The only thing I could think to do is just re-implement the GenericAvroSerde/Serializer/Deserializer classes (which are trivial) and let them grab a mock SR client out of the config. I searched around, but didn't find any better ideas "in the wild".